src/nations/nations.cpp

```diff
@@ -140,21 +141,344 @@ void restore_unsaved_values(sys::state& state) {
 	restore_cached_values(state);
 }
 
+void recalculate_markets_distance(sys::state& state) {
```

Comment:
This function probably could be parallelized if it is called sufficiently often / in the hot path. However, there are also a number of places where it could probably run in parallel as a single-threaded function to other global updates (specifically it could probably run in parallel with almost anything that doesn't alter demographics, building construction, and province ownership, from a superficial inspection).